ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListPropertiesExecutor.java

```diff
@@ -97,9 +99,17 @@ private ListPropertiesExecutor() {
         .getConfig(false)
         .getString(KsqlConfig.CONNECT_WORKER_CONFIG_FILE_PROPERTY);
 
-    return !configFile.isEmpty()
-        ? Utils.propsToStringMap(getWorkerProps(configFile))
-        : Collections.emptyMap();
+    if (configFile.isEmpty()) {
+      return Collections.emptyMap();
+    }
+
+    final Map<String, String> workerProps = Utils.propsToStringMap(getWorkerProps(configFile));
+    // only list known connect worker properties to avoid showing potentially sensitive data
+    // in other configs
+    final Set<String> allowList = new DistributedConfig(workerProps).values().keySet();
```

Comment:
This change is very contained. The only potential risk is that `DistributedConfig` has a number of required properties and this line will fail if those properties weren't set. However, the ksqlDB server should not have been able to start if `KsqlConfig.CONNECT_WORKER_CONFIG_FILE_PROPERTY` is set and a `DistributedConfig` cannot be created from it, so this should be fine. @agavra can you confirm?

Reply:
That sounds right to me - it's unfortunate that the config def is private :/ otherwise we could've just taken a look at that directly.

Reply:
Alternatively, what do you think about catching the corresponding exception and then log-and-continue? (and set the allowList to empty)

Reply:
Sure, I'm happy with that especially since we're backporting this change to older branches. I considered adding unit test coverage for this case but I don't think it's necessary; can add a quick one if you disagree.